### DISCUSSION TITLE
Convert a project down into a single file

### DISCUSSION THREAD

**Discussion by Thesnake66six** [typst/typst#4016] [Q&A]
I can't pretend to be the most fluent user of Typst, and so I may si,ply be missing something, but I've been working on a multi-file, multi-directory project that requires links that can be used between these different files. Firstly, is there a way of making these links work, and secondly, is there a good way of converting a whole project into a single file (i.e. replacing `#include` statements with the file contents) both as a workaround to this issue and as a feature in general?

**Comment by frozolotl** [comment]
I don't understand what you mean by “links” if I'm being honest.

However, I can say that there is no tool to merge a multi-file project into one. This is actually much more difficult to do reliably than it might seem at first glance.

What is the use-case for such a tool? If you just need to share your project's source code, I would recommend using ZIP files, or better, a hosted git repository.

**Reply by Thesnake66six** [reply]
Aah, sorry for not being clear - I mean using `link()` to a label created in a different file, if that makes sense. 

To be honest, I can't really think of any use cases that aren't extremely specific, or that couldn't be also done in other ways, so a tool probably won't be necessary.
